src/murmur/Messages.cpp

```diff
@@ -77,6 +77,22 @@
 		sendMessage(uSource, mppd); \
 	}
 
+#ifdef USE_GRPC
+#include <boost/fiber/all.hpp>
+#define SERVER_USER_ALIVE(user, session) \
+do { \
+		while(!qrwlVoiceThread.tryLockForRead()) { \
+			::boost::this_fiber::yield(); \
+		} \
+		if (!qhUsers.contains(session)) { \
+			qrwlVoiceThread.unlock(); \
+			return; \
+		} \
+		qrwlVoiceThread.unlock(); \
+	} while (false)
```

Comment:
What is this supposed to do?
According to https://stackoverflow.com/questions/2314066/do-whilefalse this construct is sometimes used to mimic a `goto` but I there is no `break` in the code (required for the mentioned use-case)...

Reply:
OK, this is my C programming days showing up -- it's on of my earlier commits before I had my brain in C++ mode. This is very common in C macros so that if you if you write something like

```
if (condition)
   MACRO(some, args);
else
   someOtherThing();
```

That it doesn't break your statement because it's an entire block. As to the reason behind those macros, since I'm running essentially the entire function in a fiber (as to not block the main loop if the message filter takes forever), and the implementation code happens in the main thread, *IIFC, this function is running in a server thread* I have to make sure the data isn't invalidated before the filter code runs, as well as check to make sure the sending user still exists after the signal returns.

Before, if your textMessageFilter callback took something like 3 or 4 seconds to respond, it would block the entire voice thread and there would be no audio while waiting on a RPC call that could have high latency.

Reply:
This is completely changed now, and documented. Should be a non-issue.